core/src/main/java/org/ldaptive/DnParser.java

```diff
@@ -75,26 +75,67 @@ public static String getValue(final String dn, final String name)
    * at index 0.
    *
    * @param  dn  to parse
-   * @param  index  components included in the result
+   * @param  beginIndex  components included in the result
    *
    * @return  DN from the supplied index
    *
-   * @throws  IndexOutOfBoundsException  if index is less than 0 or greater than the number of RDNs
+   * @throws  IndexOutOfBoundsException  if beginIndex is less than 0 or greater than the number of RDNs
    */
-  public static String substring(final String dn, final int index)
+  public static String substring(final String dn, final int beginIndex)
   {
-    if (index < 0) {
-      throw new IndexOutOfBoundsException("index cannot be negative");
+    if (beginIndex < 0) {
+      throw new IndexOutOfBoundsException("beginIndex cannot be negative");
     }
 
     final List<LdapAttribute> attrs = convertDnToAttributes(dn);
-    if (index >= attrs.size()) {
-      throw new IndexOutOfBoundsException("index cannot be larger than the number of RDNs");
+    if (beginIndex >= attrs.size()) {
+      throw new IndexOutOfBoundsException("beginIndex cannot be larger than the number of RDNs");
     }
 
     final StringBuilder sb = new StringBuilder();
     for (int i = 0; i < attrs.size(); i++) {
-      if (i >= index) {
+      if (i >= beginIndex) {
+        final LdapAttribute la = attrs.get(i);
+        sb.append(la.getName()).append("=").append(la.getStringValue()).append(",");
+      }
+    }
+    if (sb.length() > 0 && sb.charAt(sb.length() - 1) == ',') {
+      sb.deleteCharAt(sb.length() - 1);
+    }
+    return sb.toString();
+  }
+
+
+  /**
+   * Returns a string representation of the supplied DN beginning at beginIndex (inclusive) and ending at endIndex
+   * (exclusive). The leftmost component begins at index 0.
+   *
+   * @param  dn  to parse
+   * @param  beginIndex  components included in the result
+   * @param  endIndex  components excluded from the result
+   *
+   * @return  DN from beginIndex to endIndex
+   *
+   * @throws  IndexOutOfBoundsException  if beginIndex is less than 0, if beginIndex is greater than endIndex, or
+   * endIndex is greater than the number of RDNs
+   */
+  public static String substring(final String dn, final int beginIndex, final int endIndex)
+  {
+    if (beginIndex < 0) {
+      throw new IndexOutOfBoundsException("beginIndex cannot be negative");
+    }
+    if (beginIndex > endIndex) {
+      throw new IndexOutOfBoundsException("beginIndex cannot be larger than endIndex");
+    }
+
+    final List<LdapAttribute> attrs = convertDnToAttributes(dn);
+    if (endIndex > attrs.size()) {
+      throw new IndexOutOfBoundsException("endIndex cannot be larger than the number of RDNs");
+    }
+
+    final StringBuilder sb = new StringBuilder();
+    for (int i = 0; i < attrs.size(); i++) {
+      if (i >= beginIndex && i < endIndex) {
```

Comment:
You need to clarify the bounds on the parameters. I just checked String#substring, which works exactly like what you have done (including naming), so there's a solid precedent. You just need better javadocs on the acceptable parameter values for indices.